src/figuresClasses.ts

```diff
@@ -1,19 +1,66 @@
 export interface Figure {
-
+  shape: string;
+  color: string;
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: string = 'triangle';
 
-}
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('Invalid sides length');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const semiPerimeter = (this.a + this.b + this.c) / 2;
 
+    return Math.round((Math.sqrt(semiPerimeter * (semiPerimeter - this.a)
+      * (semiPerimeter - this.b) * (semiPerimeter - this.c))) * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: string = 'circle';
+
+  constructor(
+    public color: string,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Invalid radius length');
+    }
+  }
 
+  getArea(): number {
+    return Math.floor((Math.PI * this.radius * this.radius) * 100) / 100;
```

Comment:
You are doing same logic a lot of times(` * 100) / 100`), create function for this